examples/Folders/Readme.md

```diff
@@ -0,0 +1,45 @@
+Dropzone that accepts folders as drag-and-drop. Supports multiple folders and subfolders.
+
+```jsx harmony
+const { getDroppedOrSelectedFiles } = require('html5-file-selector')
+
+class FolderDropzone extends React.Component {
+  constructor() {
+    super()
+    this.state = { files: [] }
+  }
+
+  onDrop(files) {
+    this.setState({
+      files
+    })
+  }
+
+  render() {
+    return (
+      <section>
+        <div className="dropzone">
+          <Dropzone
+            getDataTransferItems={evt => getDroppedOrSelectedFiles(evt).then(files => files)}
```

Comment:
Since the `html5-file-selector` resolves with `files` already and this is what we expect according to https://github.com/react-dropzone/react-dropzone/pull/616/files#diff-1fdf421c05c1140f6d71444ea2b27638R97 I'm wondering if this is unnecessary to make it that explicit here.

This line can be replaced with 

```jsx
getDataTransferItems={getDroppedOrSelectedFiles}
```

On the other hand, this shows all possible hooking points for it.

I'm not sure so please express your opinion on this one. What would you prefer seeing in the example? A shorter version or this one?

Reply:
For me the explicit example is more about readability and understanding the code at once. It took me some time before I managed to understand how everything works, and my example reflects that I think. I’ve tried to make it self explanatory but of course if we explain the plugin architecture in docs, and how everything works under the hood there is no need of such explicitness.